src/main/java/org/reactome/release/dataexport/configuration/ConfigurationEntry.java

```diff
@@ -0,0 +1,261 @@
+	 *
+	 * @return Configuration value obtained from user input
+	 */
+	protected abstract String getValueFromUserInput();
+
+	/**
+	 * Gets the name or description to insert into a prompt to the user for a configuration value.
+	 *
+	 * @return Name or description to insert into a prompt to the user for a configuration value
+	 */
+	protected String getUserPromptValueName() {
+		return this.userPromptValueName;
+	}
+
+	/**
+	 * Gets the name or key of the configuration entry.
+	 *
+	 * @return Name or key of the configuration entry
+	 */
+	protected String getConfigurationEntryName() {
+		return this.configurationEntryName;
+	}
+
+	/**
+	 * Returns the full configuration entry as a String in the form of "key=value".  If the value hasn't been already
+	 * been obtained from user input via a prompt, the value will be request from the user (and cached for subsequent
+	 * calls to this method).
+	 *
+	 * @return Configuration entry as a String in the form "key=value"
+	 */
+	public String toString() {
+		if (this.userInputValue == null) {
+			this.userInputValue = getValueFromUserInput();
+		}
+
+		return getConfigurationEntryName() + "=" +  this.userInputValue;
+	}
+}
+
+/**
+ * Class to represent a configuration entry that has a value which is required from the user (i.e. there is no suitable
+ * default)
+ * @author jweiser
+ */
+class RequiredConfigurationEntry extends ConfigurationEntry {
+	private Scanner scanner;
+
+	/**
+	 * Creates an object for obtaining and representing a configuration entry.
+	 *
+	 * @param userPrompt The name or description to insert into a prompt to the user for a configuration value
+	 * @param entryName Name or key of the configuration entry (a value is obtained on request from a call
+	 * to the getValueFromUserInput() method via a user prompt)
+	 */
+	public RequiredConfigurationEntry(String userPrompt, String entryName) {
+		this(userPrompt, entryName, new Scanner(System.in));
+	}
+
+	/**
+	 * Creates an object for obtaining and representing a configuration entry.
+	 *
+	 * @param userPrompt The name or description to insert into a prompt to the user for a configuration value
+	 * @param entryName Name or key of the configuration entry (a value is obtained on request from a call
+	 * to the getValueFromUserInput() method via a user prompt)
+	 * @param scanner Scanner object from which to retrieve input for the configuration value
+	 */
+	RequiredConfigurationEntry(String userPrompt, String entryName, Scanner scanner) {
+		super(userPrompt, entryName);
+		this.scanner = scanner;
+	}
+
+	/**
+	 * Returns the value for the configuration entry.  The user is prompted for a value (with the prompt repeated if
+	 * no value is entered).
+	 *
+	 * @return Value for the configuration entry
+	 */
+	@Override
+	protected String getValueFromUserInput() {
+		String value;
+		do {
+			System.out.print("Enter " + getUserPromptValueName() + ": ");
+
+			value = getScanner().nextLine();
```

Comment:
Is this going to make it difficult to run the code from Jenkins?

Reply:
Yeah, that's right.  As long as the user doesn't specify they want to create a new configuration file and a configuration already exists and is valid, the user won't be prompted for anything.